.github/workflows/release.yml

```diff
@@ -0,0 +1,125 @@
+name: release
+
+on:
+  push:
+    tags:
+      - v*.*.*
+
+permissions:
+  contents: read
+
+jobs:
+  release-npm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Read NPM vault secrets
+        uses: hashicorp/vault-action@v2.5.0
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/jenkins-ci/npmjs/elasticmachine token | NPMJS_TOKEN ;
```

Comment:
Did you miss to put the token into the npmrc file or is this working differently?

FYI, we also have this action that takes abstracts this: https://github.com/elastic/apm-pipeline-library/tree/main/.github/actions/setup-npmrc

Reply:
Good catch, I miss this part.